.github/workflows/build.yml

```diff
@@ -270,17 +290,46 @@ jobs:
           rm -rf temp
           mv signed temp
 
-      - name: Push
+      - name: Generate VSA
+        if: env.shouldSign == 'true'
         env:
-          DOCKER_HUB_USERNAME: ${{ secrets.DOCKER_HUB_USERNAME }}
-          DOCKER_HUB_PASSWORD: ${{ secrets.DOCKER_HUB_PASSWORD }}
+          AWS_KMS_REGION:  ${{ job.build.steps.auth.with.args.aws-region }}
+          AWS_KMS_KEY_ARN: ${{ job.build.steps.sign.env.AWS_KMS_KEY_ARN }}
         run: |
-          export DOCKER_CONFIG="$PWD/.docker"
-          mkdir "$DOCKER_CONFIG"
-          trap 'find "$DOCKER_CONFIG" -type f -exec shred -fuvz "{}" + || :; rm -rf "$DOCKER_CONFIG"' EXIT
-          docker login --username "$DOCKER_HUB_USERNAME" --password-stdin <<<"$DOCKER_HUB_PASSWORD"
-          unset DOCKER_HUB_USERNAME DOCKER_HUB_PASSWORD
+          args+=(
+            --mount "type=bind,src=$PWD/signed,dst=/doi-build/signed"
+
+            # https://explore.ggcr.dev/?repo=docker/image-signer-verifier
+            "$BASHBREW_IMAGE_SIGNER"
+
+            # verify is called here to generate a VSA (https://slsa.dev/spec/v1.0/verification_summary)
+            verify
+
+            -D # debug output
+
+            --platform="$BASHBREW_ARCH"
```

Comment:
This must be a proper platform string currently, so we definitely need this to change one way or the other. 

>That being said, I'm not actually sure why this is necessary, since we're asking it to verify an image that's already a single architecture 🤔

It is due to having to handle both index manifest and image references. An image reference would only discover attestations for that image. An index manifest could have many attestations for various platform images so this arg is/was used to filter down to the specific image you are wanting to verify (as opposed to every image in the index). It is possible to make this smarter (determine if the reference is an image or index) to make the platform arg optional.

Reply:
Yeah, that's fair - a heuristic could include "does the index consist of only a single image or only a single image and the associated buildkit attestation artifact", but being explicit is probably better/safer.  In our specific case, it could be "just use the first item" and even a warning that it's doing so would be pretty reasonable, but I don't think we need to press on this specific point too hard. :+1: